kernelci/test.py

```diff
@@ -175,4 +176,19 @@ def _get_compression(url):
     params.update(plan_config.params)
     params.update(target.params)
 
+    # Scale the job priority (from 0-100) within the available levels
+    # for the lab, or use the lowest by default.
+    if 'priority' in plan_config.params:
+        priority = plan_config.params['priority']
+        if priority > 100:
+            priority = 100
+    else:
+        priority = 0
+
+    prio_range = lab_config.priority_max - lab_config.priority_min
+    priority = int(((priority * prio_range) / 100) + lab_config.priority_min)
+    params.update({
+        'priority': priority,
+    })
+
```

Comment:
Until now, priorities were treated as a LAVA-specific properties. If the lab config is not a LAVA one it won't have `priority_max` / `priority_min` defined, and other types of "labs" (local shell, k8s...) may have a very different way of managing priorities or none at all. So I think it's best to keep this in `LavaAPI.generate()` rather than here.  Or we may start having lab-specific implementations of `.get_params()`, to call the base one and then update the params dictionary, but that would mean more changes to the code.